wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -135,17 +135,26 @@ protected boolean doHandleRequest(final Request request) {
 		if (changed) {
 			// Reset validation fields
 			resetValidationState();
-			
+			// if User Model exists it will be returned, othewise Shared Model is returned
+			final FileWidgetModel sharedModel = getComponentModel();
+			// if User Model exists it will be returned, othewise it will be created
+			final FileWidgetModel userModel = getOrCreateComponentModel();
```

Comment:
To me this still looks like you are creating a component model before checking if it is needed. This is what we want you to avoid.